src/components/__tests__/integration/board.test.js

```diff
@@ -29,46 +28,51 @@ const component = (
 };
 
 describe("board responds correctly to state", () => {
-  it("finishes the game correctly", () => {
+  it("finishes the game correctly", async () => {
     const { container, getByText } = component(<Board numberOfCards={1} />);
-    fireEvent.click(container.firstChild.lastChild.firstChild);
-    fireEvent.click(container.firstChild.lastChild.lastChild);
+    const cards = container.firstChild.firstChild.nextSibling;
+    fireEvent.click(cards.firstChild);
+    fireEvent.click(cards.lastChild);
+    await waitForDomChange({ container });
     expect(getByText(/restart/i)).toBeTruthy();
   });
 
-  it("restarts the game", () => {
+  it("restarts the game", async () => {
     const { container, getByText } = component(<Board numberOfCards={1} />);
-    fireEvent.click(container.firstChild.lastChild.firstChild);
-    fireEvent.click(container.firstChild.lastChild.lastChild);
+    const cards = container.firstChild.firstChild.nextSibling;
+    fireEvent.click(cards.firstChild);
+    fireEvent.click(cards.lastChild);
+    await waitForDomChange({ container });
     fireEvent.click(getByText(/restart/i));
     expect(container).toMatchSnapshot();
   });
 
   it("updates the move counter", () => {
     const { container } = component(<Board numberOfCards={1} />);
     expect(container.firstChild.firstChild.innerHTML).toBe("0");
-    fireEvent.click(container.firstChild.lastChild.firstChild);
-    fireEvent.click(container.firstChild.lastChild.lastChild);
+    const cards = container.firstChild.firstChild.nextSibling;
+    fireEvent.click(cards.firstChild);
+    fireEvent.click(cards.lastChild);
     expect(container.firstChild.firstChild.innerHTML).toBe("1");
   });
 
   it("disables the board when 2 cards are flipped", () => {
     const { container } = component(<Board numberOfCards={2} />);
-    fireEvent.click(container.firstChild.lastChild.firstChild);
-    fireEvent.click(container.firstChild.lastChild.lastChild);
-    fireEvent.click(container.firstChild.lastChild.firstChild.nextSibling);
-    expect(
-      container.firstChild.lastChild.firstChild.nextSibling.innerHTML
-    ).toBe("");
+    const cards = container.firstChild.firstChild.nextSibling;
+    fireEvent.click(cards.firstChild);
+    fireEvent.click(cards.lastChild);
+    fireEvent.click(cards.firstChild.nextSibling);
+    expect(cards.firstChild.nextSibling.innerHTML).toBe("");
   });
 
   it("shows the card for a short span of time if incorrect", async () => {
     jest.useFakeTimers();
     const { container } = component(<Board numberOfCards={2} order={true} />);
```

Comment:
i know this change is not related to this PR, but I still dont understand what `order={true}` does..., could you please explain?

Reply:
This prop is there for the sake of testing.
When is set to true it doesn't shuffle the cards. They appear in `order`

Reply:
that logic does not make sense to me. when `order=true` there is no order? is that correct?

Reply:
No, when `order=true` it does not suffle the cards. They appear in order

Reply:
Else, the cards are shuffled randomly

Reply:
still does not make sense, shouldn't it be `shuffle=true|false`?